apps/modernization-ui/src/apps/page-builder/components/AddValueset/AddValueset.tsx

```diff
@@ -0,0 +1,161 @@
+import { ClassicButton } from '../../../../classic';
+import './AddValueset.scss';
+import { ValueSetControllerService } from '../../generated';
+import { UserContext } from '../../../../providers/UserContext';
+
+export const AddValueset = () => {
+    const { state } = useContext(UserContext);
+    // Fields
+    const [isLocalOrPhin, setIsLocalOrPhin] = useState(true);
+    const [name, setName] = useState('');
+    const [desc, setDesc] = useState('');
+    const [code, setCode] = useState('');
+
+    const [isValuesetNameNotValid, setIsValuesetNameNotValid] = useState(false);
+    const [isValuesetCodeNotValid, setIsValuesetCodeNotValid] = useState(false);
+    const [isValidationFailure, setIsValidationFailure] = useState(false);
+
+    const authorization = `Bearer ${state.getToken()}`;
+
+    const handleSubmit = () => {
+        const request = {
+            valueSetNm: name,
+            valueSetCode: code,
+            valueSetTypeCd: 'LOCAL'
+        };
+
+        ValueSetControllerService.createValueSetUsingPost({
+            authorization,
+            request
+        }).then((response: any) => {
+            setDesc('');
+            setCode('');
+            setName('');
+            return response;
+        });
+        // xox- POST API call here
+        // POST /page-builder/api/v1/valueset/
+    };
+    const handleClose = () => {
+        window.close();
+    };
+
+    const handleBack = () => {
+        history.go(-1);
+    };
+    const validateValuesetName = (name: string) => {
+        const pattern = /^[a-zA-Z0-9_]*$/;
+        if (name.match(pattern)) {
+            setIsValuesetNameNotValid(false);
+            setIsValidationFailure(false);
+        } else {
+            setIsValuesetNameNotValid(true);
+            setIsValidationFailure(true);
+        }
+    };
+
+    const validateValuesetCode = (name: string) => {
+        const pattern = /^[a-zA-Z0-9_]*$/;
+        if (name.match(pattern)) {
+            setIsValuesetCodeNotValid(false);
+            setIsValidationFailure(false);
+        } else {
+            setIsValuesetCodeNotValid(true);
+            setIsValidationFailure(true);
+        }
+    };
+
+    return (
+        <div className="add-valueset">
+            <Button className="usa-button--unstyled close-btn" type={'button'} onClick={handleClose}>
+                <Icon.Close />
+            </Button>
+            <h3 className="main-header-title" data-testid="header-title">
+                <Button className="usa-button--unstyled back-btn" type={'button'} onClick={handleBack}>
+                    <Icon.ArrowBack />
+                </Button>
+                Add value set
+            </h3>
+            <div className="add-valueset__container">
+                <h3 className="header-title">Add value set</h3>
+                <p className="fields-info">
+                    All fields with <span className="mandatory-indicator">*</span> are require
```

Comment:
should be `required` instead of `require`


and can we disable the submit (`Add to question`) button until required fields are populated

Reply:
Sure